src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/WorkspaceDiagnosticRefreshTest.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.Hosting;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.AspNetCore.Razor.Test.Common.LanguageServer;
+using Microsoft.AspNetCore.Razor.Test.Common.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Moq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test;
+
+public class WorkspaceDiagnosticRefreshTest(ITestOutputHelper testOutputHelper) : LanguageServerTestBase(testOutputHelper)
+{
+    [Fact]
+    public async Task WorkspaceRefreshSent()
+    {
+        var projectSnapshotManager = CreateProjectSnapshotManager();
+        var clientConnection = new StrictMock<IClientConnection>();
+        clientConnection
+            .Setup(c => c.SendNotificationAsync(Methods.WorkspaceDiagnosticRefreshName, It.IsAny<CancellationToken>()))
+            .Returns(Task.CompletedTask)
```

Comment:
Does `ReturnsAsync` not work here?

Reply:
It looks like not. `ReturnsAsync` doesn't have a way to do a `Task`, it requires `Task<T>`

Reply:
Ah, I think I'm remembering a Moq extension I had added in a branch to address that.

Reply:
> Idk tbh. What's the difference?

`ReturnsAsync(...)` is just convenient for returning `Task<T>` and `ValueTask<T>` in mocks.